torch/distributed/fsdp/ssd_offload.py

```diff
@@ -0,0 +1,958 @@
+        ssd_handle += 1
+        ssd_handle.to_file()
+        ssd_handle2 = torch.load('/home/user/checkpoint.pkl')
+
+        print(f"handles are equal: {torch.equals(ssd_handle, ssd_handle2)}")
+
+    One would expect this to print False, however unintuitively it will print True.
+    ssd_handle.filename and ssd_handle2.filename are equal. This means that
+    when we execute torch.load, we read from the .pkl file and write the result into
+    /home/user/handle.bin, clobbering the updated result from `ssd_handle += 1`
+
+    We want to give the user the possibility of not clobbering the data using this
+    Context Manager.
+
+        ssd_handle = SsdTensorHandle.from_tensor(ref_tensor)
+        ssd_handle.set_file_params('/home/user/handle.bin', 0)
+        with CheckpointPathContextManager(override_path='/home/user/checkpoint_data/'):
+            torch.save(ssd_handle, '/home/user/checkpoint.pkl')
+        ssd_handle += 1
+        ssd_handle.to_file()
+        ssd_handle2 = torch.load('/home/user/checkpoint.pkl')
+
+        print(f"handles are equal: {torch.equals(ssd_handle, ssd_handle2)}")
+
+    This code results with ssd_handle.filename = '/home/user/handle.bin' and ssd_handle2.filename =
+    `/home/user/checkpoint_data/handle.bin'. Therefore the torch.load won't clobber ssd_handle, and
+    the printed result is False.
+
+    """
+
+    def __init__(self, override_path: str) -> None:
+        self.old_path = SsdTensorHandle.override_directory_path
+        self.override_path = override_path
+
+    def __enter__(self) -> None:
+        SsdTensorHandle.override_directory_path = self.override_path
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_value: Optional[BaseException],
+        exec_traceback: Optional[TracebackType],
+    ) -> None:
+        SsdTensorHandle.override_directory_path = self.old_path
+
+
+class SsdParameter(SsdTensorHandle, torch.nn.Parameter):
+    @classmethod
+    def from_tensor(cls: Type[SsdParameter], tensor: SsdTensorHandle) -> SsdParameter:  # type: ignore
+        r = cls(tensor.shape, tensor.dtype, tensor.requires_grad, device=tensor.device)
+        r.point_to_tensor(tensor)
+        return r
+
+    @staticmethod
+    def __new__(
+        cls: Type[SsdParameter],
+        shape: torch.Size,
+        dtype: torch.dtype,
+        requires_grad: bool = True,
+        device: torch.device = torch.device("cpu"),
+    ) -> SsdParameter:
+        r = super(SsdParameter, cls).__new__(cls, shape=shape, dtype=dtype, requires_grad=requires_grad, device=device)
+        return r  # type: ignore
+
+    def __init__(
+        self,
+        shape: torch.Size,
+        dtype: torch.dtype,
+        requires_grad: bool = True,
+        device: torch.device = torch.device("cpu"),
+    ) -> None:
+        super(SsdParameter, self).__init__(shape=shape, dtype=dtype, requires_grad=requires_grad, device=device)
+
+
+class SsdFlatParameter(SsdParameter):
```

Comment:
once #79652 is landed, define a SsdFlatParameter on top of it?

Reply:
Once it that PR has landed, we can re-evaluate how to best integrate this. If .data usage is still happening, we will may have to define SsdFlatParameter as a FlatParamHandle subclass.